src/Microsoft.Management.Configuration/ConfigurationSetParser_0_2.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ConfigurationSetParser_0_2.h"
+
+#include <AppInstallerErrors.h>
+#include <AppInstallerStrings.h>
+
+#include <sstream>
+
+namespace winrt::Microsoft::Management::Configuration::implementation
+{
+    using namespace AppInstaller::YAML;
+
+#define FIELD_TYPE_ERROR(_field_,_mark_) SetError(WINGET_CONFIG_ERROR_INVALID_FIELD_TYPE, (_field_), (_mark_)); return
+#define FIELD_TYPE_ERROR_IF(_condition_,_field_,_mark_) if (_condition_) { FIELD_TYPE_ERROR(_field_,_mark_); }
+
+#define FIELD_VALUE_ERROR(_field_,_value_,_mark_) SetError(WINGET_CONFIG_ERROR_INVALID_FIELD_VALUE, (_field_), (_mark_), (_value_)); return
+#define FIELD_VALUE_ERROR_IF(_condition_,_field_,_value_,_mark_) if (_condition_) { FIELD_VALUE_ERROR(_field_,_value_,_mark_); }
+
+    namespace
+    {
+        // Contains the qualified resource name information.
+        struct QualifiedResourceName
+        {
+            QualifiedResourceName(hstring input)
+            {
+                std::wstring_view inputView = input;
+                size_t pos = inputView.find('/');

```

Comment:
should we fail if there are more than one / ?

Reply:
Given that `/` isn't a valid character in module or resource names, I don't think I mind if we break non-functional files at a later date.